active/003-seq-feature.md

```diff
@@ -141,10 +156,18 @@ f.update((1, 2))
 `__del__`
 - Deletes a `BoundFeature`.  The corresponding interval(s) in the `IntervalTree` will be removed when this is called.
 
-## `IntervalMetadata` functions
+
+## `IntervalMetadata` object structure
+### Attributes
+* `features`: a list of `BoundFeature` object
+* `_update_flag`: boolean. Whenever any intervals of any member feature is modified, this is set to True, indicating `_intervaltree` needs to be updated.
+* `_intervaltree` (implemented as property): `IntervalTree` object created from all the intervales in the `features`. When this attributes is accessed by any methods (like `query` below), it checks `_update_flag` to decide whether to re-create from all intervals before it returns the interval tree. This lazy update saves computation.
```

Comment:
It will work with `weakref`. Here is a simplified example:

```
$ import weakref

$ class BoundFeature:
|     def __init__(self, wr=None):
|         self.wr = wr

$ a = BoundFeature()

$ b = BoundFeature()

$ class IM:
|    def __init__(self, fl=None):
|        for f in fl:
|            f.wr = weakref.ref(self)
|        self.features = fl

$ im = IM([a, b])

$ im.features
: 
[<__main__.BoundFeature at 0x113cd5a58>,
 <__main__.BoundFeature at 0x113cd5f28>]

$ im.features[0].wr
: <weakref at 0x113cecae8; to 'IM' at 0x113cd7c88>
```